tests/src/JsonResponse.php

```diff
@@ -15,9 +15,9 @@ class JsonResponse extends Response
     public function __construct($status = 200, array $headers = [], $body = null, $version = '1.1', $reason = null)
     {
         if (isset($body)) {
-            if (is_file($body)) {
+            if (is_string($body) && is_file($body)) {
```

Comment:
How could $body be string and file at the same time?

Reply:
It can't, but the method is suppose to allow you to pass one of:

1. A file name in the `fixtures` folder.
1. A raw JSON string.
1. An array of data to automatically json_encode.

The lack of an `is_string` check was breaking the third case as an array was being passed into `is_file`.

Reply:
But still, I don't see how this if could be ever be true, if is_file($body) were true as we expected, is_string($body) would be false, so it could never enter that condition...

Reply:
Oh, the first if actually checks for any file, such as an absolute path. It's the second if that takes a raw file name and looks in the fixtures directory.

Reply:
Ah ok! Sorry for that, so a file is actually a string and a file path, so is_file() returns true. I've should have checked the is_file() documentation before asking this question.